exampleCourse/questions/element/hiddenHints/server.py

```diff
@@ -0,0 +1,14 @@
+# Hint is Shown if the user clicks Save & Grade
+
+
+def grade(data):
+    if data["submitted_answers"]["c_2"] == 35:
+        data["feedback"]["show_hint_35"] = True
+
+
+# Hint is Shown if the user clicks Save only
+
+
+def parse(data):
+    if data["submitted_answers"]["c_2"] == 53:
```

Comment:
You shouldn't do this in the parse function, as it will display the hint even if Save Only is hit (and it should only really update on Save & Grade). The case in the grade function is what is recommended for most people.

Reply:
The reason a second example was given in a `parse()` function is because @jonatanschroeder suggested that there were two places that one code provide hints (see #12015). One might want to give a hint if the Save Only button is hit.

Reply:
I kind of agree with @eliotwrobson here, though I think a comment could point out the possibility of using `parse` instead. One major point about this is the fact that `pl-hint` already takes *valid* submission counts as an option for conditionally displaying hints, and this count is only incremented after grading the submission. So, for consistency, it should be the default that any conditional for hints should also follow the same logic.

Reply:
> I kind of agree with @/eliotwrobson here, though I think a comment could point out the possibility of using `parse` instead.

Agree here. People copy and paste from the example course a lot, so it's important to try and demonstrate what is useful in most situations.